Add information about different types of ACS (#1971)
theforeman/foreman-documentation: guides/common/modules/proc_configuring-custom-alternate-content-sources.adoc

[id="Configuring_Custom_Alternate_Content_Sources_{context}"]
= Configuring Custom Alternate Content Sources

.Prerequisites
* If the repository requires SSL authentication, import the SSL certificate and key to the {Project}.
For more information, see {ContentManagementDocURL}Importing_Custom_SSL_Certificates_content-management[Importing Custom SSL Certificates] in _{ContentManagementDocTitle}_. 

> * Note that the alternate content source paths consist of a base URL appended with the subpaths that you provide. For example, if your base URL is `https://server.example.com` and your subpaths are `rhel7/` and `rhel8/`, then both `https://server.example.com/rhel7/` and `https://server.example.com/rhel8/` will be searched.

Reviewer: ```suggestion
* Note that the alternate content source paths consist of a base URL appended with the subpaths that you provide.
For example, if your base URL is `https://server.example.com` and your subpaths are `rhel7/` and `rhel8/`, then both `https://server.example.com/rhel7/` and `https://server.example.com/rhel8/` will be searched.
```

[id="Configuring_Custom_Alternate_Content_Sources_{context}"]
= Configuring Custom Alternate Content Sources

.Prerequisites
* If the repository requires SSL authentication, import the SSL certificate and key to the {Project}.
For more information, see {ContentManagementDocURL}Importing_Custom_SSL_Certificates_content-management[Importing Custom SSL Certificates] in _{ContentManagementDocTitle}_. 
* Note that the alternate content source paths consist of a base URL appended with the subpaths that you provide.

> For example, if your base URL is `https://server.example.com` and your subpaths are `rhel7/` and `rhel8/`, then both `https://server.example.com/rhel7/` and `https://server.example.com/rhel8/` will be searched.

Reviewer: ```suggestion
For example, if your base URL is `https://{foreman-example-com}` and your subpaths are `rhel7/` and `rhel8/`, then both `https://{foreman-example-com}/rhel7/` and `https://{foreman-example-com}/rhel8/` will be searched.
```

Only apply this if you mean "Foreman/Satellite server" with "server".

Reviewer 2: The server in this case will be something external to the Foreman.

Reviewer: Then please ignore this comment!

[id="Configuring_Custom_Alternate_Content_Sources_{context}"]
= Configuring Custom Alternate Content Sources

.Prerequisites

> * If the repository requires SSL authentication, import the SSL certificate and key to the Satellite.

Reviewer: ```suggestion
* If the repository requires SSL authentication, import the SSL certificate and key to the your {ProjectServer}.
```

I am unsure what's really required as content credential. Is is just the certificate? See https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#Importing_Custom_SSL_Certificates_content-management

Author: @maximiliankolb let me check with @ianballou once.

Author: @ianballou could you help with this?

Reviewer 2: I'm not really understanding the exact question, but I'll lay it out:

SSL client cert, SSL client key, and SSL CA cert are set via content credentials.

If the upstream repo requires certificate authentication, use a client cert and key.

If you verify SSL, you can optionally include a CA certificate (again, as a content credential). If you have the upstream CA in your CA trust store, no need to include a CA certificate.

Reviewer: Are we talking about something else than "content credentials within Foreman", which, to my understanding, always only contain the SSL certitficate as documented? Are we talking about adding things manually to the trust store of Foreman Server?

Reviewer 2: It's just Foreman Content Credentials like you say, which can be ssl client certs, ssl client keys, or ssl ca certs (anything SSL Certificate related).

We don't need to mention adding anything to Foreman's trust store since the user can supply a CA certificate instead.
